subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -177,9 +196,9 @@ int nrf_cloud_encode_sensor_data(const struct nrf_cloud_sensor_data *sensor,
 		return -ENOMEM;
 	}
 
-	ret = json_add_str(root_obj, "appId", sensor_type_str[sensor->type]);
-	ret += json_add_str(root_obj, "data", sensor->data.ptr);
-	ret += json_add_str(root_obj, "messageType", "DATA");
+	ret = json_add_str_cs(root_obj, JSON_KEY_APPID, sensor_type_str[sensor->type]);
+	ret += json_add_str_cs(root_obj, JSON_KEY_DATA, sensor->data.ptr);
+	ret += json_add_str_cs(root_obj, JSON_KEY_MSGTYPE, MSGTYPE_VAL_DATA);
```

Comment:
in this case, i don't think it is necessary to check each individual line and add an error/return.
the parameters are already checked, so a failure would be due to out of memory, correct?

Reply:
Looks like something else is up with the return value of this function. `json_add_str_cs` can return `-EINVAL`, `-ENOMEM`, `NULL` and the pointer to the pointer to the allocated JSON string? Looking at the `cJSON_AddStringToObjectCS` which is called under the hood.

Reply:
`json_add_str_cs` returns `-EINVAL` for an invalid parameter and `-ENOMEM` if `cJSON_AddStringToObjectCS` returns a NULL pointer.

in this case, the parameters have already been validated. 

in this file, i don't think checking each call (for the json building) provides much troubleshooting assistance; but if you have found some instances where it would, let's fix it.

Reply:
Lets leave it as is. :smile: